docs/DdnRelocatableShaderElf.md

```diff
@@ -0,0 +1,205 @@
+# Unlinked Shaders And Relocatable Shader Elf
+
+# INTRODUCTION
+
+There is interest in being able to compile SPIR-V shaders without having all of
+the pipeline state information available.  This is advantagous when the same
+shader is used in multiple pipelines because it will only be compiled once.  It
+is also possible to compile the shaders "offline", that is without running the
+full vulkan application. Unlinked and relocatable shaders were designed
+with the intend of allowing offline compilation of shaders that will be linked
+together during pipeline creation.
+
+# Unlinked shaders vs relocatable shaders
+
+The two concepts are similar, but there is a minor, and important, distinction.
+An unlinked shader is an LGC concept.  LGC will generate an unlinked shader
+when the option in the pipeline context is set. See `m_unlinked` in
+[llpcPipelineContext.cpp](https://github.com/GPUOpen-Drivers/llpc/blob/dev/llpc/context/llpcPipelineContext.h).
+When possible, LGC will generate code that can be linked with other unlinked
+shader and the pipeline state to create the pipeline.  The link step is
+suppose to be reletively fast, and it implemented in the `ElfLinkerImpl` class
+in
+[ElfLinker.cpp](https://github.com/GPUOpen-Drivers/llpc/blob/dev/lgc/elfLinker/ElfLinker.cpp).
+
+A relocatable shader is an LLPC concept.  The flag `enableRelocatableShaderElf`
+in the [pipeline
+options](https://github.com/GPUOpen-Drivers/llpc/blob/dev/include/vkgcDefs.h)
+causes LLPC to use relocatable shaders when building the
+pipeline. When building with relocatable shaders, LLPC will create a new pipeline
+state object for each shader, stripping out as much as it can from the
+pipeline state.  The new pipeline is passed to LGC, where it will be compiled as
+an unlinked shader.  LLPC will then link all of those unlinked shaders together
+to create the pipeline.
+
+The key difference is that unlinked shaders are built with whatever information
+they have while relocatable shaders will remove anything that can be handled
+when linking.
```

Comment:
This could use an example.  After reading this, I'm not really sure about the difference between unlinked and relocatable shaders.  Perhaps show how relocations are needed because no pipeline state is present (as is the case with unlinked shaders)?

Reply:
I've added something.  Let me know if it helps.